feat: add missing batch repositories

jackhodgkiss opened the pull request.

sjpb commented on ansible/inventory/group_vars/all/source-repositories:
```diff
@@ -455,6 +455,27 @@ source_repositories:
+      - codeowners:
+          content: "{{ community_files.codeowners.batch }}"
+          dest: ".github/CODEOWNERS"
+  slurm_slurm_appliance:
```
Is this supposed to be `ansible-slurm-appliance` c.f change in terraform/github/terraform.tfvars.json?

jackhodgkiss replied:
Fixed the typo.

sjpb commented on terraform/github/terraform.tfvars.json:
```diff
@@ -48,7 +48,8 @@
       "ansible-slurm-appliance",
-      "caas-slurm-appliance"
+      "caas-slurm-appliance",
+      "slurm_image_builder"
```
See above.

sjpb reviewed (changes requested):
the batch team is probably also a good owner for https://github.com/stackhpc/ansible-role-cluster-nfs

jackhodgkiss commented on ansible/inventory/group_vars/all/source-repositories:
```diff
@@ -455,6 +455,27 @@ source_repositories:
+      - codeowners:
+          content: "{{ community_files.codeowners.batch }}"
+          dest: ".github/CODEOWNERS"
+  slurm_image_builder:
```
Removed `slurm_image_builder`.

jackhodgkiss commented:
Added `ansible-role-cluster-nfs` and removed `slurm_image_builder`. If this is good to go I just need to run an import workflow before this gets merged.

markgoddard commented on ansible/inventory/group_vars/all/source-repositories:
```diff
@@ -448,13 +448,34 @@ source_repositories:
           dest: ".github/CODEOWNERS"
+  caas-slurm-appliance:
+    repository_type: "ansible"
+    workflows: "{{ ansible_workflows.role }}"
```
This isn't an Ansible role, so this workflow isn't appropriate. An empty list should be fine here.

markgoddard commented on ansible/inventory/group_vars/all/source-repositories:
```diff
@@ -448,13 +448,34 @@ source_repositories:
+          dest: ".github/CODEOWNERS"
+  ansible-slurm-appliance:
+    repository_type: "ansible"
+    workflows: "{{ ansible_workflows.role }}"
```
ditto
